File: shared/src/main/java/ru/privatenull/pnauth/extension/AuthExtensionRegistry.java
package ru.privatenull.pnauth.extension;
import java.util.Optional;
import java.util.UUID;
import java.util.concurrent.CompletionStage;
import java.util.function.Consumer;
public interface AuthExtensionRegistry {
    AuthExtensionRegistration register(String id, int priority, AuthPolicyHook hook);
    CompletionStage<AuthPolicyDecision> evaluate(AuthOperationContext context);
    Optional<VerificationTicket> pending(UUID uniqueId);
    boolean approve(String ticketId);
    boolean deny(String ticketId);
    AuthExtensionRegistration onTicket(Consumer<VerificationTicket> listener);
}

File: shared/src/main/kotlin/ru/privatenull/pnauth/extension/AuthExtensionRegistry.kt
package ru.privatenull.pnauth.extension

import java.util.Optional
import java.util.UUID
import java.util.concurrent.CompletionStage
import java.util.function.Consumer

interface AuthExtensionRegistry {
    fun register(id: String, priority: Int, hook: AuthPolicyHook): AuthExtensionRegistration
    fun evaluate(context: AuthOperationContext): CompletionStage<AuthPolicyDecision>
    fun pending(uniqueId: UUID): Optional<VerificationTicket>
    fun approve(ticketId: String): Boolean
    fun deny(ticketId: String): Boolean
    fun onTicket(listener: Consumer<VerificationTicket>): AuthExtensionRegistration
}
